engine.go

```diff
@@ -593,14 +593,7 @@ func (e *Engine) analyzeNode(ctx *sql.Context, query string, bound sql.Node, qFl
 		// todo(max): improve name resolution so we can cache post name-binding.
 		// this involves expression memoization, which currently screws up aggregation
 		// and order by aliases
-		prepStmt, _, err := e.Parser.ParseOneWithOptions(ctx, query, sqlMode.ParserOptions())
```

Comment:
any gut instinct on whether the comment is outdated?

Reply:
User vars did not work previously in Stored Procedures, but they work better after the rewrite.
It seems like we do cache, so the TODO is outdated(?). Can delete.